src/Tests/IQsharpEngineTests.cs

```diff
@@ -713,6 +713,21 @@ await Assert.That
                 .UsingEngine()
                 .Input("%lsmagi")
                 .ExecutesWithError(containing:
+                    // NB: Since %attach is only present in local development
+                    //     mode, hints may in general change between debug
+                    //     and release configuration.
+                    #if DEBUG
```

Comment:
Nit:
I don't know this par of C# yet, but if it is the same as a C/C++ preprocessor then consider 
`#ifdef DEBUG` or `#if defined(DEBUG)` (`ifdef` instead of `if`)
or a standard `#ifndef NDEBUG` or `#if !defined(NDEBUG)` (`NDEBUG` is a standard identifier in C/C++, as opposed to `DEBUG`).

Reply:
The syntax in C# is similar to, but not exactly the same as, that used by the C/C++ preprocessor. In particular, `#if` in C# is similar to `#ifdef` in C and C++: https://docs.microsoft.com/dotnet/csharp/language-reference/preprocessor-directives#conditional-compilation